scripts/db_writer.py

```
@@ -135,9 +140,60 @@ def _upsert_repo(cur, repo: Dict, platform: str):
+                    FROM repos
+                    ON CONFLICT (repo_id, snapshot_date) DO UPDATE SET
+                        stars = EXCLUDED.stars,
+                        download_count = EXCLUDED.download_count,
+                        forks = EXCLUDED.forks,
+                        open_issues = EXCLUDED.open_issues,
+                        latest_release_date = EXCLUDED.latest_release_date,
+                        captured_at = NOW()
+                """)
+                count = cur.rowcount
+        print(f"  ✓ daily snapshot: {count} repos captured for today (UTC)")
+    except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)